### PR TITLE
[ Flatten ] forwarding / copy for Flatten Layer @open sesame 06/16 13:29

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR provides the fowarding/copy function of Flatten Layer
. implement forwarding function
. implement copy function

Resolves:

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/src/flatten_layer.cpp [hunk 1]
```diff
@@ -23,13 +23,32 @@ namespace nntrainer {
...
+  hidden = Tensor(output_dim);
+
   return status;
 }
 
 Tensor FlattenLayer::forwarding(Tensor in, int &status) {
-  // NYI
-  return in;
+  memcpy(hidden.getData(), in.getData(),
+         in.getDim().getDataLen() * sizeof(float));
```
Comment (dongju-chae) [inline nntrainer/src/flatten_layer.cpp:49]: How about adding new method `getDataSize()` to `Tensor`?

Comment (jijoongmoon) [reply nntrainer/src/flatten_layer.cpp:49]: right. I will do it another PR. Thanks.

**Comment by jijoongmoon** [comment]
Yes. I aware of that. But each layer handles their own memory for now. I'm not for sure if it is safe or not. So I thought once we check everything is going to be alright, then we could delete deep memory copy between inter-layers. Especially copy includes here, 
```    
for (unsigned int i = layers.size() - 1; i > 0; i--) {                        
    X = layers[i]->forwarding(X, iteration);                                 
  }  
```

**Comment by kparichay** [comment]
Great, the let's go forward with this PR.

**Pull request merged**